extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpPermissionChecker.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import java.util.concurrent.CompletionStage;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+
+public interface HttpPermissionChecker {
+
+    /**
+     * The priority of the permission checker. Permission checkers are evaluated from highest to lowest
+     * priority.
+     *
+     * @return The priority
+     */
+    int getPriority();
```

Comment:
@stuartwdouglas I'm not sure this is required, all the registered checkers have to give their approval, the order in which they do it is likely to be insignificant

Reply:
With the current implementation they don't. They can return PERMIT which grants access, or IGNORE, which will grant access if subsequent checkers don't deny (and default-deny is false)

Reply:
May be worth having a default implementation of this method ? I'm not sure one can reliably predict the ordering by returning `100` or something like that :-)

Reply:
No, in fact I should throw an error if we have multiple checkers with the same priority. The main thing that we need is deterministic ordering, if multiple checkers have the same priority then it could result in non-deterministic behaviour.

Reply:
Hi Stuart, it feels things are being over complicated. How can multiple modules can make it deterministic ? Each module has its own expectations what it needs to check. For example, with the Config converters it is a different story, but here, 

If I have to ship my custom checker I need to make sure it is invoked, so I need to be aware what are the actual numbers the other modules are bringing in, otherwise if I just put random 100 then there is no guarantee my checker will even be invoked.

IMHO all the checkers have to be invoked as opposed to building some relationship between them, it is either PERMIT-ALL or 403.
Can you think of the practical case where a PERMIT-ALL mode can be too lax or too restrictive, etc ?
Thanks